tests/unit/features/partitions/test_lifecycle_manager.py

```diff
@@ -0,0 +1,138 @@
+from hypothesis import HealthCheck, given, settings, strategies
+
+from craft_parts import errors
+from craft_parts.lifecycle_manager import LifecycleManager
+
+
+class TestPartitionsSupport:
+    """Verify LifecycleManager supports partitions."""
+
+    @pytest.fixture
+    def parts_data(self) -> Dict[str, Any]:
+        return {"parts": {"foo": {"plugin": "nil"}}}
+
+    @pytest.mark.parametrize("partitions", [["default"], ["default", "kernel"]])
+    def test_project_info(self, new_dir, parts_data, partitions):
+        """Verify partitions are parsed and passed to ProjectInfo."""
+        lifecycle = LifecycleManager(
+            parts_data,
+            application_name="test_manager",
+            project_name="project",
+            cache_dir=new_dir,
+            work_dir="work_dir",
+            arch="aarch64",
+            parallel_build_count=16,
+            custom="foo",
+            partitions=partitions,
+        )
+        info = lifecycle.project_info
+
+        assert info.application_name == "test_manager"
+        assert info.project_name == "project"
+        assert info.target_arch == "arm64"
+        assert info.arch_triplet == "aarch64-linux-gnu"
+        assert info.parallel_build_count == 16
+        assert info.dirs.parts_dir == new_dir / "work_dir" / "parts"
+        assert info.dirs.stage_dir == new_dir / "work_dir" / "stage"
+        assert info.dirs.prime_dir == new_dir / "work_dir" / "prime"
+        assert info.custom_args == ["custom"]
+        assert info.custom == "foo"
+        assert info.partitions == partitions
+
+    @pytest.mark.parametrize("partitions", [None, []])
+    def test_no_partitions(self, new_dir, parts_data, partitions):
+        """Raise an error if the partitions feature is enabled but not defined."""
+        with pytest.raises(errors.FeatureError) as raised:
+            LifecycleManager(
+                parts_data,
+                application_name="test_manager",
+                cache_dir=new_dir,
+                partitions=partitions,
+            )
+
+        assert (
+            raised.value.message
+            == "Partition feature is enabled but no partitions are defined."
+        )
+
+    @pytest.mark.parametrize(
+        "partitions", [["defaulta"], ["kernel"], ["kernel", "default"]]
+    )
+    def test_default_not_first(self, new_dir, parts_data, partitions):
+        """Raise an error if the first partition is not 'default'."""
+        with pytest.raises(ValueError) as raised:
+            LifecycleManager(
+                parts_data,
+                application_name="test_manager",
+                cache_dir=new_dir,
+                partitions=partitions,
+            )
+
+        assert str(raised.value) == "First partition must be 'default'."
+
+    @settings(suppress_health_check=[HealthCheck.function_scoped_fixture])
+    @given(
+        partitions=strategies.lists(
+            strategies.text(strategies.sampled_from([*ascii_lowercase]), min_size=1),
```

Comment:
`sampled_from` will take any ordered collection, of which a string is one, so you don't technically need to convert it to a list.

Since an empty string is presumably an invalid partition name (`min_size=1` tipped me off), it might be good to test for a partition set of `["default", ""]`

Reply:
I improved the strategy and added a test for `["default", ""]`